### PR TITLE
Feat: Ruby SDK

### PR DESCRIPTION
**Pull request by DanielHougaard**
This PR introduces a new Ruby SDK. For the Ruby SDK, we've decided to make the authentication happen through individual authentication methods, rather than in the client constructor.

This is a design structure we're going to roll out for the other SDK's eventually. A date for the full rollout is yet to be decided. Having individual auth methods makes the SDK's more flexible, and more inline with the design of our standalone Go SDK.

### CODE REVIEW

File: .github/workflows/release-ruby.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
+name: Release Ruby SDK
```
Comment (maidul98) [inline .github/workflows/release-ruby.yml:1]: I don't see us supporting FreeBSD and Windows 32 (older but people could still be on it)

File: crates/infisical/src/manager/auth/kubernetes.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    client: &mut Client,
+    input: &KubernetesAuthMethod,
+) -> Result<AccessTokenSuccessResponse> {
+    return kubernetes_login(
+        client,
+        input.identity_id.clone(),
+        input
+            .service_account_token_path
+            .clone()
+            .unwrap_or("/var/run/secrets/kubernetes.io/serviceaccount/token".to_string()),
```
Comment (maidul98) [inline crates/infisical/src/manager/auth/kubernetes.rs:16]: nit: Isn't this already a string?

**Review by maidul98 — APPROVED** [review]
Don't know Ruby that well but skimmed though it and it looks like it should work

File: .github/workflows/release-ruby.yml (hunk 1 above)
Comment (DanielHougaard) [reply .github/workflows/release-ruby.yml:1]: True, we should add FreeBSD -- I'll put this down as a todo.

We do support Windows x86 (32-bit) though. The naming is a bit strange, but our Windows library is cross compatible with both x86/x64

**Pull request merged**